### PR TITLE
Fix category filter bug by caching decoded categories

### PR DESCRIPTION
**Pull request by davidmanzanares**

### CODE REVIEW

File: src/renderer/viz/expressions/ViewportFeatures.js [hunk 1]
```diff
@@ -78,6 +78,7 @@ export default class ViewportFeatures extends BaseExpression {
 
     accumViewportAgg(feature) {
         const properties = this._getMinimumNeededSchema().columns;
+        console.log('!!!', { feature, properties });
```
Comment (davidmanzanares) [inline src/renderer/viz/expressions/ViewportFeatures.js]: hey!

Comment (elenatorro) [reply src/renderer/viz/expressions/ViewportFeatures.js]: 🤦‍♂️

File: src/renderer/viz/expressions/ViewportFeatures.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ export default class ViewportFeatures extends BaseExpression {
         return this.expr;
     }
 
-    resetViewportAgg() {
+    _resetViewportAgg() {
```
Comment (IagoLast) [inline src/renderer/viz/expressions/ViewportFeatures.js]: _resetViewportAgg is called from the outside so I think it should be public

**Comment by davidmanzanares** [comment]
Hey, as we discussed off-channel there are some important issues with this PR. Since I think we should push the fix as soon as possible I'm gonna create a new PR with the initial commit and a single regression test.

If you think that some of the changes here can be re-used, feel free to create a new PR, but please, take the mentioned issues (API breaking changes, performance regression) into account.

**Pull request closed without merging**